packages/cdk/resources/Cloudfront/CloudfrontFunction.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Function,
+  FunctionCode,
+  FunctionRuntime,
+  IKeyValueStore,
+  ImportSource,
+  KeyValueStore
+} from "aws-cdk-lib/aws-cloudfront"
+import {Construct} from "constructs"
+import {readFileSync} from "fs"
+import {resolve} from "path"
+
+interface CodeReplacement {
+  valueToReplace: string
+  replacementValue: string
+}
+type codeReplacements = Array<CodeReplacement>
+
+interface KeyValue {
+  key: string
+  value: string
+}
+type KeyValues = Array<KeyValue>
+
+export interface CloudfrontFunctionProps {
+  readonly sourceFileName: string
+  readonly codeReplacements?: codeReplacements
+  readonly keyValues?: KeyValues
+}
+
+/**
+ * Cloudfront function with support for KVS and code replacement
+
+ */
+
+export class CloudfrontFunction extends Construct {
+  public readonly function: Function
+
+  public constructor(scope: Construct, id: string, props: CloudfrontFunctionProps){
+    super(scope, id)
+
+    // Resources
+    let functionStore: IKeyValueStore | undefined = undefined
+    if (props.keyValues){
+      functionStore = new KeyValueStore(this, "FunctionsStore", {
+        source: ImportSource.fromInline(JSON.stringify({data: props.keyValues}))
+      })
+    }
+
+    /* Automatically include replacement of export statements as not supported by CF Functions,
+    and replace placeholder KVS ID if KVS is required */
+    const codeReplacements: codeReplacements = [
+      {valueToReplace: "export", replacementValue: ""},
+      ...functionStore ? [
+        {valueToReplace: "KVS_ID_PLACEHOLDER", replacementValue: functionStore.keyValueStoreId}] : [],
+      ...props.codeReplacements ?? []
+    ]
+    let functionCode = readFileSync(
+      resolve(__dirname, `../../../cloudfrontFunctions/src/${props.sourceFileName}`), "utf8")
+    for (const codeReplacement of codeReplacements){
+      functionCode = functionCode.replace(codeReplacement.valueToReplace, codeReplacement.replacementValue)
+    }
+
+    const cloudfrontFunction = new Function(this, "Function", {
+      code: FunctionCode.fromInline(functionCode),
```

Comment:
can we add a functionName param to this as it makes it so much easier to debug rather than trying to figure out what eu-west-2stateless-resouratusCodeFunction8775FC94 is or what stack it belongs to